### PR TITLE
Use true random string as uri for public calendars - as a result we c…

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description

A true random uri for public calendars is required

@LukasReschke as discussed
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
Makes sense :+1: 

I assume this is backward compatible because old uris are already in the DB ? @DeepDiver1975

**Comment by DeepDiver1975** [comment]
> I assume this is backward compatible because old uris are already in the DB ?

No idea about backwards compat - but I don't care much: this is a 9.2 feature

File: apps/dav/lib/CalDAV/Publishing/PublishPlugin.php [hunk 1]
```diff
@@ -94,22 +94,16 @@ public function initialize(Server $server) {
...
 		if ($node instanceof Calendar) {
-			$token = md5($this->config->getSystemValue('secret', '').$node->getResourceId());
-
-			$publishUrl = $this->urlGenerator->getAbsoluteURL($this->server->getBaseUri().'public-calendars/').$token;
-
-			$propFind->handle('{'.self::NS_CALENDARSERVER.'}publish-url', function () use ($node, $publishUrl) {
+			$propFind->handle('{'.self::NS_CALENDARSERVER.'}publish-url', function () use ($node) {
 				if ($node->getPublishStatus()) {
 					// We return the publish-url only if the calendar is published.
+					$token = $node->getName();
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Publishing/PublishPlugin.php]: $node->getName() returns the "good" name of the calendar, not the token.

Comment (DeepDiver1975) [inline apps/dav/lib/CalDAV/Publishing/PublishPlugin.php]: damn

**Pull request merged**